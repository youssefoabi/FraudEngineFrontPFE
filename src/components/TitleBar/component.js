import { isNil, mergeDeepLeft } from 'ramda';
import { TitleBar as MUITitleBar } from '@cdiscount/backoffice-ui';
import { shape } from 'prop-types';
import React from 'react';
import ReactRouterPropTypes from 'react-router-prop-types';

function TitleBar({ history, location, ...props }) {
  const isNotHome = !isNil(history) && !isNil(location) && location.pathname !== '/';

  const enhancedProps = isNotHome
    ? mergeDeepLeft(props, {
        backButtonProps: {
          onClick: history.goBack,
        },
      })
    : props;

  return <MUITitleBar {...enhancedProps} />;
}

TitleBar.propTypes = {
  history: shape({
    pathname: ReactRouterPropTypes.location.pathname,
  }),
  location: shape({
    goBack: ReactRouterPropTypes.history.goBack,
  }),
};

TitleBar.defaultProps = {
  history: null,
  location: null,
};

export default TitleBar;
