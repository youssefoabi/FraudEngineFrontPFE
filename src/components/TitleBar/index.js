import { withRouter } from 'react-router-dom';

import TitleBar from './component';

export default withRouter(TitleBar);
