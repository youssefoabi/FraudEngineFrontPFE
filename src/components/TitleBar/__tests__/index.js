import { itRendersAllMutations } from '@cdiscount/jest-utils';

import TitleBar from '../component';

describe(TitleBar.name, () => {
  itRendersAllMutations(TitleBar, [
    { name: 'without props', props: null },
    {
      name: 'with location.pathname and history.goBack',
      props: {
        location: { pathname: '/foo' },
        history: { goBack: jest.fn() },
      },
    },
  ]);
});
