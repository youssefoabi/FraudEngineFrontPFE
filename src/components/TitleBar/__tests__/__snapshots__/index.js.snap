// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TitleBar TitleBar renders correctly with location.pathname and history.goBack 1`] = `
<f
  actions={Array []}
  backButtonProps={
    Object {
      "onClick": [MockFunction],
    }
  }
  dividerProps={
    Object {
      "color": "transparent",
      "size": 3,
    }
  }
  title={null}
/>
`;

exports[`TitleBar TitleBar renders correctly without props 1`] = `
<f
  actions={Array []}
  backButtonProps={
    Object {
      "icon": [Function],
      "label": "Back",
      "onClick": null,
    }
  }
  dividerProps={
    Object {
      "color": "transparent",
      "size": 3,
    }
  }
  title={null}
/>
`;
